docs/elements.md

```diff
@@ -311,19 +311,46 @@ This element supports additional preview options through [element extensions](el
 
+Note that there is a file size limit of **5 MB per answer**. This limit is not customizable as larger
+requests will be rejected by the server. For the same reason, it is also not possible to bypass the
+limit by using multiple `pl-file-upload` elements in the same question. To avoid unexpected errors or
+potentially misleading error messages for large file uploads, we recommend not using more than one
+`pl-file-upload` element per question.
+
 #### Sample element
 
 ![](elements/pl-file-upload.png)
 
 ```html
-<pl-file-upload file-names="foo.py, bar.c, filename with\, comma.txt"></pl-file-upload>
+<pl-file-upload
+  file-names="foo.py, bar.c, filename with\, comma.txt"
+  optional-file-names="foo.pdf, *.py, file_?.txt, file_[abc].txt, file with \\? as non-wildcard.pdf"
+></pl-file-upload>
 ```
 
 #### Customizations
 
-| Attribute    | Type     | Default | description                                                                                                                                    |
-| ------------ | -------- | ------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
-| `file-names` | CSV list | ""      | List of files that should and must be submitted. Commas in a filename should be escaped with a backslash, and filenames cannot contain quotes. |
+| Attribute             | Type     | Default | description                                                                                                                                                                                                                                                       |
+| --------------------- | -------- | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `file-names`          | CSV list | ""      | List of files that should and must be submitted. Commas in a filename should be escaped with a backslash, and filenames cannot contain quotes.                                                                                                                    |
+| `optional-file-names` | CSV list | ""      | List of files that can be submitted, but are optional. Wildcards are supported (see below). Commas should be escaped with a backslash, reserved wildcard characters in file names should be escaped with double backslashes, and filenames cannot contain quotes. |
+| `allow-blank`         | boolean  | false   | Whether or not an empty file input is allowed. By default, at least one file must be uploaded, even if only `optional-file-names` are specified                                                                                                                   |
```

Comment:
Should allow-blank just be removed entirely (and effectively set to always-on)? I don't disagree that this makes the element a bit simpler and only removes support for a rather specific corner case. Especially if I add the pattern option as proposed below, it might be good to simplify here.

Reply:
I think the issue then is that you can't make a blank submission with only required files without this attribute? It might be better to just allow the optionals to be blank unconditionally.

Reply:
I think the _real_ issue is that `optional-file-names` is used as both a new attribute to support wildcards without backwards-compat concerns, and also to support optional files if `allow-blank` is specified. I'd think about this in the context of the below comment where we discuss possibly changing how "allow wildcards" is achieved.

Reply:
I've made an update to remove ```allow-blank```. I think no matter what the outcome of the discussion below is, the only use case we eliminate by removing the parameter (and essentially treating it as set to true) is the case where someone wants to allow optional files, but not an empty submission. 

I'm thinking of a case like ```allow-blank="false" optional-file-names="solution.txt,solution.pdf"``` to give students the choice which file format they want to use. Without ```allow-blank```, this would also allow empty submissions. However, even with ```allow-blank```, it also allows submissions of both files at once, so there's always some extra (manual or automated) sanity checking necessary, so I think we are not losing much here.